### ISSUE TITLE
pr #3088 made some commands not work..

### ISSUE THREAD

**Issue by Shane-XB-Qian** [fatih/vim-go#3093]
in pr #3088 limits some command away -if 'go#package#InGOPATH()' was not true.
// e.g ':GoCallees' ':GoCallstack' etc not working..
but actually such commands were not so strict limited by gopath, it can be working in the past..

**Comment by bhcleek** [comment]
The change you're referring to was in #2704, because `guru` does support modules. I'm reluctant to re-enable it for that reason, but I'll give it some thought.

**Comment by Shane-XB-Qian** [comment]
i saw you check-in those all at `CommitDate: Thu Nov 12 20:56:58 2020 -0800`, 
and yes, looks you are limiting for `guru`, looks those commands almost all depends on `guru`, but those commands are useful, even was not in gopath or something.. simply directly using them probably had benefits...
// those not like https://github.com/fatih/vim-go/issues/3030 `GoKeyify`, keyify looks had to wait gopls to make it work again.

**Comment by bhcleek** [comment]
While some `guru` commands might work in module aware mode in limited circumstances, because it cannot be counted on to work reliably, I do not intend to enable the commands outside of GOPATH mode. If you find guru useful, I would encourage you to continue to use GOPATH mode so that you will have the guru commands available.

**Comment by Shane-XB-Qian** [comment]
not sure why you are saying it is not 'reliably', it had worked outside of gopath long time,
or can we have alternative way from gopls to make those commands work again?
// at least let's prompt it 's not supported if out of gopath -vs- now it's just saying `E492 not a editor command`.

**Comment by bhcleek** [comment]
Guru does not work reliably outside of GOPATH, because it _relies_ on GOPATH and is not aware of modules at all.

I am working to convert as many of the commands that relied on guru to use gopls as possible.

**Comment by Shane-XB-Qian** [comment]
um.. really? mostly my proj -or just a single tmp gofile it's not in gopath, even had no go-mod, but i recall i had used e.g `GoChannelPeers` long time..
// if you say so, then just wish them came back soon.. though i do not like this limitation.. and as i know most golang proj now is not out of gopath.. so such limitation almost saying those commands not working perhaps for ever if so..

**Comment by bhcleek** [comment]
While I've read many comments that guru doesn't work with modules and I know for sure that it uses `go/build` instead of `golang.org/x/tools/packages`, it does seem to work in a many cases. I did some testing locally and found that most of it's functions work with modules in most circumstances.

I've created #3096 to re-enable the guru commands until I have a better understanding of the situations in which guru does not work with modules.
